photon-ml/src/main/scala/com/linkedin/photon/ml/avro/data/DataProcessingUtils.scala

```diff
@@ -117,9 +117,24 @@ object DataProcessingUtils {
     } else {
       1.0
     }
+
+    val metaMap = Utils.getMapAvro(record, AvroFieldNames.META_DATA_MAP)
     val ids = randomEffectIdSet
       .map { randomEffectId =>
-        (randomEffectId, Utils.getStringAvro(record, randomEffectId))
+        val idValue = Utils.getStringAvro(record, randomEffectId, isNullOK = true)
+        val finalIdValue = if (idValue == null) {
```

Comment:
I think there is an integTest related with these part of changes. The previous non-nullable change actually failed it. Sorry about that, I'll see whether I could add a branch there to make it covered.